web/models.py

```
@@ -2195,47 +2210,60 @@ def __str__(self):
 
     def complete_quiz(self):
         """Mark the quiz as completed and calculate final score."""
         from django.utils import timezone
 
         self.completed = True
         self.end_time = timezone.now()
-        self.calculate_score()
         self.save()
 
+    @property
+    def max_score(self):
```

Ruff (0.8.2)

Missing return type annotation for public function `max_score`

(ANN201)